tests/validator_test.go

```diff
@@ -329,4 +338,618 @@ var _ = Describe("Template validator", func() {
+				"name.os.template.kubevirt.io/fedora26": "Fedora 26",
+				"name.os.template.kubevirt.io/fedora27": "Fedora 27",
+				"name.os.template.kubevirt.io/fedora28": "Fedora 28",
+				"validations":                           validation,
+			},
+			Labels: map[string]string{
+				"os.template.kubevirt.io/fedora26":      "true",
+				"os.template.kubevirt.io/fedora27":      "true",
+				"os.template.kubevirt.io/fedora28":      "true",
+				"workload.template.kubevirt.io/generic": "true",
+				"flavor.template.kubevirt.io/small":     "true",
+				"template.kubevirt.io/type":             "base",
+			},
+		},
+		Parameters: []templatev1.Parameter{
+			{
+				Description: "VM name",
+				From:        "fedora-[a-z0-9]{16}",
+				Generate:    "expression",
+				Name:        "NAME",
+			},
+			{
+				Name:        "PVCNAME",
+				Description: "Name of the PVC with the disk image",
+				Required:    true,
+			},
+		},
+	}
+
+	codec := serializer.NewCodecFactory(kubevirtv1.Scheme).LegacyCodec(kubevirtv1.GroupVersion)
+	template.Objects = append(template.Objects,
+		runtime.RawExtension{
+			Raw: []byte(runtime.EncodeOrDie(codec, &kubevirtv1.VirtualMachine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "${NAME}",
+					Namespace: strategy.GetNamespace(),
+					Labels: map[string]string{
+						"vm.kubevirt.io/template": "fedora-desktop-small",
+						"app":                     "${NAME}",
+					},
+				},
+				Spec: kubevirtv1.VirtualMachineSpec{
+					Running: &running,
+					Template: &kubevirtv1.VirtualMachineInstanceTemplateSpec{
+						Spec: kubevirtv1.VirtualMachineInstanceSpec{
+							Domain: kubevirtv1.DomainSpec{
+								CPU: &kubevirtv1.CPU{
+									Sockets: 1,
+									Cores:   1,
+									Threads: 1,
+								},
+								Resources: kubevirtv1.ResourceRequirements{
+									Requests: map[core.ResourceName]resource.Quantity{
+										"memory": resource.MustParse("2Gi"),
+									},
+								},
+								Devices: kubevirtv1.Devices{
+									Rng: &kubevirtv1.Rng{},
+									Disks: []kubevirtv1.Disk{
+										{
+											Name: "rootdisk",
+											DiskDevice: kubevirtv1.DiskDevice{
+												Disk: &kubevirtv1.DiskTarget{
+													Bus: "virtio",
+												},
+											},
+										},
+									},
+								},
+							},
+							EvictionStrategy: &liveMigrate,
+							Volumes: []kubevirtv1.Volume{
+								{
+									Name: "rootdisk",
+									VolumeSource: kubevirtv1.VolumeSource{
+										PersistentVolumeClaim: &core.PersistentVolumeClaimVolumeSource{
+											ClaimName: "${PVCNAME}",
+										},
+									},
+								},
+								{
+									Name: "cloudinitvolume",
+									VolumeSource: kubevirtv1.VolumeSource{
+										CloudInitNoCloud: &kubevirtv1.CloudInitNoCloudSource{
+											UserData: userData,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			})),
+		})
+	return template
+}
+
+func TemplateWithRules() templatev1.Template {
+	validations := `[
```

Comment:
can you please use rule struct (e.g. https://github.com/kubevirt/kubevirt-template-validator/blob/1c714f043f18b5957bf3ebebce0226e756adbb75/pkg/validation/rules.go#L32) and then use json.Marshal() to serialize it.

Reply:
Will do it in the part when we move template-validator here in the ssp-operator